regexing/src/main/java/Extraction.java

```diff
@@ -40,6 +54,13 @@
      */
     public List<String> extractEmails(String input){
         List<String> emails = new ArrayList<String>();
+        Pattern p = Pattern.compile("\\s(\\w+(?:\\.\\w+)?@\\w+\\.\\w+)\\s");
```

Comment:
While this works for the tests this won't work in the situation of Dr Who<doctor@who.com>.  You are only checking for a space.  This is a time where word boundary `\b` would be more effecient.